hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/DiskBalancerManager.java

```diff
@@ -107,7 +108,20 @@ public List<HddsProtos.DatanodeDiskBalancerInfoProto> getDiskBalancerStatus(
     List<DatanodeDetails> filterDns = null;
     if (hosts.isPresent() && !hosts.get().isEmpty()) {
       filterDns = NodeUtils.mapHostnamesToDatanodes(nodeManager, hosts.get(),
-          useHostnames);
+          useHostnames).stream()
+          .filter(dn -> {
+            try {
+              if (nodeManager.getNodeStatus(dn) != NodeStatus.inServiceHealthy()) {
+                LOG.warn("Datanode {} is not in optimal state for disk balancing." +
+                    " Operational state: {}", dn.getHostName(), nodeManager.getNodeStatus(dn).getOperationalState());
```

Comment:
I'm not sure this log is needed? If yes, please change to:
```suggestion
          .filter(dn -> {
            try {
              NodeStatus nodeStatus = nodeManager.getNodeStatus(dn);
              if (nodeStatus != NodeStatus.inServiceHealthy()) {
                LOG.warn("Datanode {} is not in optimal state for disk balancing." +
                    " Operational state: {}", dn.getHostName(), nodeStatus.getOperationalState());
```
otherwise it would query the nodeStatusMap twice

Reply:
I recommend keeping the LOG.warn statement for the case where datanodes are explicitly specified by the user as users can get clarity from logs that status is not displayed due to DN not in service or healthy.